src/os/bluestore/BlueStore.cc

```diff
@@ -13974,6 +13986,76 @@ int BlueStore::_deferred_replay()
   return r;
 }
 
+bool BlueStore::_eliminate_outdated_deferred(bluestore_deferred_transaction_t* deferred_txn,
```

Comment:
I would definitely like to have some (preferable unit) test coverage for this function implementation. But given the urgency and @aclamk limited availability at the moment - we can stick with using deep-fsck in run_test_deferred.sh for now. See my suggestion for the relevant commit which adds this script.